packages/grafana-ui/src/components/FilterPill/FilterPill.tsx

```diff
@@ -7,23 +7,48 @@ import { useStyles2 } from '../../themes';
 import { IconName } from '../../types';
 import { clearButtonStyles } from '../Button';
 import { Icon } from '../Icon/Icon';
+import { Tooltip, TooltipPlacement } from '../Tooltip';
 
 export interface FilterPillProps {
   selected: boolean;
-  label: string;
+  label: React.ReactNode;
   onClick: React.MouseEventHandler<HTMLElement>;
   icon?: IconName;
+  className?: string;
+  tooltip?: string;
+  tooltipPlacement?: TooltipPlacement;
 }
 
-export const FilterPill = ({ label, selected, onClick, icon = 'check' }: FilterPillProps) => {
+export const FilterPill = ({
+  label,
+  selected,
+  onClick,
+  icon = 'check',
+  className,
+  tooltip,
+  tooltipPlacement,
+}: FilterPillProps) => {
   const styles = useStyles2(getStyles);
   const clearButton = useStyles2(clearButtonStyles);
-  return (
-    <button type="button" className={cx(clearButton, styles.wrapper, selected && styles.selected)} onClick={onClick}>
+  const pill = (
+    <button
+      type="button"
+      className={cx(clearButton, styles.wrapper, selected && styles.selected, className)}
+      onClick={onClick}
+    >
       <span>{label}</span>
       {selected && <Icon name={icon} className={styles.icon} />}
     </button>
   );
+
+  if (tooltip) {
+    return (
+      <Tooltip content={tooltip} placement={tooltipPlacement}>
+        {pill}
+      </Tooltip>
+    );
+  }
```

Comment:
Can you not just wrap this component in tooltip yourself? We favor composability over configuration here.

Reply:
I can but I thought we use a tooltip in anything clickable, which pill is and so thought this was an inconsistency.